ddtrace/commands/ddtrace_run.py

```diff
@@ -105,6 +104,15 @@ def main():
 
     log.debug("program executable: %s", executable)
 
+    if os.path.basename(executable) == "uwsgi":
+        print(
+            (
+                "ddtrace-run is not supported with uwsgi. "
+                "See https://ddtrace.readthedocs.io/en/stable/advanced_usage.html#uwsgi for details."
+            )
+        )
+        sys.exit(1)
```

Comment:
I don't think we can have this here. There are people using `ddtrace-run` with uwsgi, they are going to upgrade and then their deployments will be broken.

Reply:
Which is good since they will be able to fix it.
Silently fails and making the application half-broken is not a better choice IIUC.

Reply:
I think my argument is that this example works, no issues with producing and flushing traces.

```
ddtrace-run uwsgi --http 0.0.0.0:8000 --wsgi app:app
```


Can we detect if someone is using processes/master/etc and then fail hard with a warning message?

Can we detect if they aren't using the example setup we need (threads enabled, etc)?

Reply:
> Can we detect if they aren't using the example setup we need (threads enabled, etc)?

This is done in #1992 but it also has some limitations (and it not ready to be merged).

If we're not happy with that, maybe the idea is to remove the `sys.exit` and then wait for #1992 to improve things which leaves the "issue" open until then.

Reply:
I think the `sys.exit` here really scares me. We need a _really_ good reason to add it.

I am worried about someone who happens to be on the happy path who does a deploy and now their server won't run, and they have to dig through logs to find this link, and then they have to update their deployment scripts or config... hoping they are the team which owns the process, etc.

I think the risks are really high here.

Reply:
I went with `sys.exit` here to do more than loudly complain since this will cause stability issues. My conclusion was that it's worse for a customer to continue experiencing stability issues than a breaking change in the upgrade. We could instead just loudly complain but we'll have most uwsgi customers miss the need to change their configuration.